src/escape/escape.c

```diff
@@ -270,6 +279,18 @@ static int run(int argc, char *argv[]) {
                 if (i != argv + optind)
                         fputc(' ', stdout);
 
+                if (arg_hashed) {
+                        if (strlen(e) >= UNIT_NAME_MAX) {
+                                _cleanup_free_ char *n = NULL;
+
+                                r = unit_name_hash_long(e, &n);
+                                if (r < 0)
+                                        return log_error_errno(r, "Failed to hash name: %m");
+
+                                free_and_replace(e, n);
+                        }
+                }
+
```

Comment:
this smells like something we should always apply, i.e. not be opt-in? i mean, pid1 doesnt ask you nicely either...

Reply:
I am just worried that someone there might be using systems-escape for something other than generating a name for a unit and they don't care about the lenght.

Reply:
well, i'd be worried more about the people who used it for its intended purpose and then end up with escapings that aren't actually how systemd calls its units?

Reply:
oook. I will do it that way.

Reply:
Yeah, I think it makes sense to do it this way.